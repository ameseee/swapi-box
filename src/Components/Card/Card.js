import React from 'react';
import PropTypes from 'prop-types';
import '../../index.css';

const Card = ({ toggleFavorited, ...attributes}) => {
  const attributeKeys = Object.keys(attributes);

  const cardContent = attributeKeys.map(attributeKey => {
    return <h4 value={attributes[attributeKey]} key={attributeKey}>{attributeKey}: {attributes[attributeKey]}</h4>;
  });

  return (
    <div>
      <article
        className='card unfavorited'
        onClick={toggleFavorited}>
        {cardContent}
        <span className="card-background"></span>
      </article>
    </div>
  );
};

Card.propTypes = {
  vehicleName: PropTypes.string,
  model: PropTypes.string,
  passengers: PropTypes.string,
  vehicleClass: PropTypes.string,
  personName: PropTypes.string,
  homeworld: PropTypes.string,
  population: PropTypes.string,
  species: PropTypes.string,
  planetName: PropTypes.string,
  terrain: PropTypes.string,
  climate: PropTypes.string,
  residents: PropTypes.array,
  selected: PropTypes.string,
  toggleFavorited: PropTypes.func
};

export default Card;
