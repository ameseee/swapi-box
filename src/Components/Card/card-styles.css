.card {
  margin: 4px;
  background: black;
  width: 360px;
  height: auto;
  padding: 17px;
  padding-bottom: 15px;
  font-weight: 100;
  font-size: 17px;
  color: #e5e5e5;
  position: relative;
  background: linear-gradient(182deg, #596980, #916386, #916a75, #568f84);
  background-size: 800% 800%;
  -webkit-animation: AnimationName 15s ease infinite;
  animation: AnimationName 15s ease infinite; }

@-webkit-keyframes AnimationName {
  0% {
    background-position: 50% 0%; }
  50% {
    background-position: 51% 100%; }
  100% {
    background-position: 50% 0%; } }

@keyframes AnimationName {
  0% {
    background-position: 50% 0%; }
  50% {
    background-position: 51% 100%; }
  100% {
    background-position: 50% 0%; } }

.card-background {
  background: black;
  opacity: .5; }

.favorited {
  background-color: #4d6d88; }

.info-title {
  font-weight: bold; }

button {
  color: #e5e5e5;
  background-color: #000;
  border: none;
  padding: 4px; }

.fave-btn {
  height: 20px;
  position: fixed;
  top: 10px;
  right: 10px; }
