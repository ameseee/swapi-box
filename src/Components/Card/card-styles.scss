.card {
  margin: 4px;
  background: black;
  width: 360px;
  height: auto;
  padding: 17px;
  padding-bottom: 15px;
  font-weight: 100;
  font-size: 17px;
  color: #e5e5e5;

  background: linear-gradient(182deg, #596980, #916386, #916a75, #568f84);
  background-size: 800% 800%;

  -webkit-animation: AnimationName 15s ease infinite;
  animation: AnimationName 15s ease infinite;

  @-webkit-keyframes AnimationName {
      0%{background-position:50% 0%}
      50%{background-position:51% 100%}
      100%{background-position:50% 0%}
  }
  @keyframes AnimationName {
      0%{background-position:50% 0%}
      50%{background-position:51% 100%}
      100%{background-position:50% 0%}
  }
}

.card-background{
  background: black;
  opacity: .5;
}

.favorited {
  border: 2px solid magenta;
}

.info-title {
  font-weight: bold;
}

button {
  color: #fff;
  background-color: #000;
  width: 20px;
  border: none;
  padding: 2px;
}
