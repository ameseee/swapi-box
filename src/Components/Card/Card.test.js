import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import config from '../../setupTests.js';

describe('CARD', () => {
  const mkFun = jest.fn();
  let wrapper = shallow(<Card />);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should render a containing div', () => {
    const div = wrapper.find('div');

    expect(div.length).toEqual(1);
  });

  it('should render a card article', () => {
    const div = wrapper.find('.card');

    expect(div.length).toEqual(1);
  });


  it('should match the Card snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

});
