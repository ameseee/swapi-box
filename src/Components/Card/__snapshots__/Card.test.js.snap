// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CARD should match the Card snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": "LOADING IN CARD",
    },
    "ref": null,
    "rendered": "LOADING IN CARD",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "LOADING IN CARD",
      },
      "ref": null,
      "rendered": "LOADING IN CARD",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
